scripts/dex-router/src/index.ts

```diff
@@ -0,0 +1,294 @@
+import { getPdaAuthority } from "@kamino-finance/limo-sdk/dist/utils";
+import { getVersionedTxSize } from "./utils/size";
+import { LOOKUP_TABLE_ADDRESS, OPPORTUNITY_WAIT_TIME } from "./const";
+
+export class DexRouter {
+  private client: Client;
+  private pingTimeout: NodeJS.Timeout | undefined;
+  private connectionSvm: Connection;
+  private expressRelayConfig?: ExpressRelaySvmConfig;
+  private readonly routers: Router[];
+  private readonly executor: Keypair;
+  private readonly chainId: string;
+  private readonly PING_INTERVAL = 30000;
+
+  constructor(
+    endpoint: string,
+    skExecutor: string,
+    chainId: string,
+    endpointSvm: string
+  ) {
+    this.client = new Client(
+      {
+        baseUrl: endpoint,
+      },
+      undefined,
+      this.opportunityHandler.bind(this),
+      this.bidStatusHandler.bind(this)
+    );
+    this.routers = [new JupiterRouter()];
+    this.executor = Keypair.fromSecretKey(
+      anchor.utils.bytes.bs58.decode(skExecutor)
+    );
+    this.chainId = chainId;
+    this.connectionSvm = new Connection(endpointSvm, "confirmed");
+  }
+
+  async bidStatusHandler(bidStatus: BidStatusUpdate) {
+    let resultDetails = "";
+    if (bidStatus.type == "submitted" || bidStatus.type == "won") {
+      resultDetails = `, transaction ${bidStatus.result}`;
+    } else if (bidStatus.type == "lost") {
+      if (bidStatus.result) {
+        resultDetails = `, transaction ${bidStatus.result}`;
+      }
+    }
+    console.log(
+      `Bid status for bid ${bidStatus.id}: ${bidStatus.type}${resultDetails}`
+    );
+  }
+
+  async opportunityHandler(opportunity: Opportunity) {
+    console.log("Received opportunity:", opportunity.opportunityId);
+    await new Promise((resolve) => setTimeout(resolve, OPPORTUNITY_WAIT_TIME));
+    try {
+      const bid = await this.generateRouterBid(opportunity as OpportunitySvm);
+      const result = await this.client.requestViaWebsocket({
+        method: "post_bid",
+        params: {
+          bid: bid,
+        },
+      });
+      if (result === null) {
+        throw new Error("Empty response in websocket for bid submission");
+      }
+      console.log(
+        `Successful bid. Opportunity id ${opportunity.opportunityId} Bid id ${result.id}`
+      );
+    } catch (error) {
+      console.error(
+        `Failed to bid on opportunity ${opportunity.opportunityId}: ${error}`
+      );
+    }
+  }
+
+  async generateRouterBid(opportunity: OpportunitySvm): Promise<{
+    transaction: string;
+    chain_id: string;
+  }> {
+    const order = opportunity.order;
+    const clientLimo = new limo.LimoClient(
+      this.connectionSvm,
+      order.state.globalConfig
+    );
+    const inputMintDecimals = await clientLimo.getOrderInputMintDecimals(order);
+    const outputMintDecimals = await clientLimo.getOrderOutputMintDecimals(
+      order
+    );
+
+    let routerOutputs: RouterOutput[] = [];
+    for (const router of this.routers) {
```

Comment:
nit (since we only have a single router ATM):a better way to do this is via Promise.all + map